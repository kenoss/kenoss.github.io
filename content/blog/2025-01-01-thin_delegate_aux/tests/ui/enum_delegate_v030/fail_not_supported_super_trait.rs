use enum_delegate_v030 as enum_delegate;

#[enum_delegate::delegate]
trait ShapeI: std::fmt::Debug {
    fn area(&self) -> f64;
}

#[derive(Debug)]
#[enum_delegate::delegate(derive(ShapeI))]
enum Shape {
    Rect(Rect),
    Circle(Circle),
}

#[derive(Debug)]
struct Rect {
    width: f64,
    height: f64,
}

impl ShapeI for Rect {
    fn area(&self) -> f64 {
        self.width * self.height
    }
}

#[derive(Debug)]
struct Circle {
    radius: f64,
}

impl ShapeI for Circle {
    fn area(&self) -> f64 {
        3.14 * self.radius * self.radius
    }
}

fn main() {}
