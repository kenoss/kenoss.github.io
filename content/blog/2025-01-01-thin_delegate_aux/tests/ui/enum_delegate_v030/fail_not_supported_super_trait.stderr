error[E0277]: `OnlyMarkerSelfBoundsSupportedForNow` doesn't implement `Debug`
 --> tests/ui/enum_delegate_v030/fail_not_supported_super_trait.rs:3:1
  |
3 | #[enum_delegate::delegate]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^ `OnlyMarkerSelfBoundsSupportedForNow` cannot be formatted using `{:?}`
  |
  = help: the trait `Debug` is not implemented for `OnlyMarkerSelfBoundsSupportedForNow`
  = note: add `#[derive(Debug)]` to `OnlyMarkerSelfBoundsSupportedForNow` or manually `impl Debug for OnlyMarkerSelfBoundsSupportedForNow`
note: required by a bound in `assert_impl_all`
 --> tests/ui/enum_delegate_v030/fail_not_supported_super_trait.rs:4:15
  |
3 | #[enum_delegate::delegate]
  | -------------------------- required by a bound in this function
4 | trait ShapeI: std::fmt::Debug {
  |               ^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the attribute macro `enum_delegate::delegate` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `OnlyMarkerSelfBoundsSupportedForNow` with `#[derive(Debug)]`
  |
3 + #[derive(Debug)]
4 | #[enum_delegate::delegate]
  |
